pkg/planner/core/memtable_infoschema_extractor.go

```diff
@@ -661,6 +672,13 @@ func findTableAndSchemaByName(
 		schemaSlice = append(schemaSlice, st.schema)
 		tableSlice = append(tableSlice, st.table)
 	}
+	sort.Slice(schemaSlice, func(i, j int) bool {
```

Comment:
We only need to sort by schema name?

Reply:
https://github.com/pingcap/tidb/pull/56378/commits/291d8cc58a967e9e2f13c710f2573ec712fe4b1c